backend/typescript/services/implementations/__tests__/checkInService.test.ts

```diff
@@ -0,0 +1,120 @@
+import CheckIn from "../../../models/checkIn.model";
+import User from "../../../models/user.model";
+import Volunteer from "../../../models/volunteer.model";
+
+import testSql from "../../../testUtils/testDb";
+import {
+  testCheckIns,
+  testVolunteersDb,
+  testUpdatedCheckIns,
+  testUsersDb,
+} from "../../../testUtils/checkInService";
+import nodemailerConfig from "../../../nodemailer.config";
+import IEmailService from "../../interfaces/emailService";
+import EmailService from "../emailService";
+import CheckInService from "../checkInService";
+import VolunteerService from "../volunteerService";
+import IVolunteerService from "../../interfaces/volunteerService";
+
+jest.mock("nodemailer", () => {
+  const createTransport = jest.fn().mockReturnValue({
+    sendMail: jest.fn(),
+  });
+  return { createTransport };
+});
+
+describe("pg checkInService", () => {
+  let checkInService: CheckInService;
+
+  beforeEach(async () => {
+    await testSql.sync({ force: true });
+    const emailService: IEmailService = new EmailService(nodemailerConfig);
+    const volunteerService: IVolunteerService = new VolunteerService();
+    checkInService = new CheckInService(emailService, volunteerService);
+    await User.bulkCreate(testUsersDb);
+    await Volunteer.bulkCreate(testVolunteersDb);
+    await CheckIn.bulkCreate(testCheckIns);
+  });
+
+  afterAll(async () => {
+    await testSql.sync({ force: true });
+    await testSql.close();
+  });
+
+  it("create single day CheckIn", async () => {
+    // pass in the id of a user (1)
+    const mockCreateCheckInDTO = {
+      userId: "1",
+      startDate: new Date("2021-09-01T09:00:00.000Z"),
+      endDate: new Date("2021-09-01T00:10:00.000Z"),
+      volunteerId: "1",
```

Comment:
I feel like for a more realistic test scenario, we should have it be similar to what the request body would be from the FE UI:
```suggestion
      startDate: new Date("2021-09-01T09:00:00.000Z"),
      endDate: new Date("2021-09-01T00:10:00.000Z"),
```